homeassistant/components/homeconnect/api.py

```diff
@@ -0,0 +1,389 @@
+            elif app.type == "Dishwasher":
+                device = Dishwasher(app)
+            elif app.type == "FridgeFreezer":
+                device = FridgeFreezer(app)
+            elif app.type == "Oven":
+                device = Oven(app)
+            elif app.type == "CoffeeMaker":
+                device = CoffeeMaker(app)
+            elif app.type == "Hood":
+                device = Hood(app)
+            elif app.type == "Hob":
+                device = Hob(app)
+            else:
+                _LOGGER.warning("Appliance type %s not implemented.", app.type)
+                continue
+            devices.append({"device": device, "entities": device.get_entities()})
+        return devices
+
+
+class HomeConnectDevice:
+    """Generic Home Connect device."""
+
+    # for some devices, this is instead 'BSH.Common.EnumType.PowerState.Standby'
+    # see https://developer.home-connect.com/docs/settings/power_state
+    power_off_state = "BSH.Common.EnumType.PowerState.Off"
+
+    def __init__(self, appliance):
+        """Initialize the device."""
+        self.appliance = appliance
+        try:
+            self.appliance.get_status()
+        except (HomeConnectError, ValueError):
+            _LOGGER.debug("Unable to fetch appliance status. Probably offline.")
+        try:
+            self.appliance.get_settings()
+        except (HomeConnectError, ValueError):
+            _LOGGER.debug("Unable to fetch settings. Probably offline.")
+        try:
+            program_active = self.appliance.get_programs_active()
+        except (HomeConnectError, ValueError):
+            _LOGGER.debug("Unable to fetch active programs. Probably offline.")
+            program_active = None
+        if program_active and "key" in program_active:
+            self.appliance.status["BSH.Common.Root.ActiveProgram"] = {
+                "value": program_active["key"]
+            }
+        self.appliance.listen_events(callback=self.event_callback)
+        self.entities = []
+
+    def event_callback(self, appliance):
+        """Handle event."""
+        _LOGGER.debug("Update triggered on %s", appliance.name)
+        _LOGGER.debug(self.entities)
+        _LOGGER.debug(self.appliance.status)
+        for entity in self.entities:
+            entity.async_entity_update()
+
+
+class HomeConnectEntity(Entity):
+    """Generic Home Connect entity (base class)."""
+
+    def __init__(self, device, name):
+        """Initialize the entity."""
+        self.device = device
+        self._name = name
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def name(self):
+        """Return the name of the node (used for Entity_ID)."""
+        return self._name
+
+    @property
+    def unique_id(self):
+        """Return the unique id base on the id returned by Home Connect and the entity name."""
+        return f"{self.device.appliance.haId}-{self.name}"
```

Comment:
We should just use the haId, as names can change.

Reply:
A type would be ok.

Reply:
I had that initially, but then I thought the `unique_id` has to be unique at the entity level, not device level (which is what the `haId` is). Am I misunderstanding?

Reply:
Bump.

Reply:
OK, just to be sure I understand correctly. At the moment the unique ID has the form, e.g.
`SomeID123-Oven Door`
`SomeID456-Dryer Door`
`SomeID456-Dryer LaundryCare.Dryer.Program.Cotton`

Are you suggesting that I remove the `Oven`/`Dryer` part?